tests/unit/test_solvers/test_solution.py

```diff
@@ -268,11 +276,36 @@ def test_save(self):
             solution.save_data("test.csv", to_format="csv")
         # only save "c" and "2c"
         solution.save_data("test.csv", ["c", "2c"], to_format="csv")
+        csv_str = solution.save_data(variables=["c", "2c"], to_format="csv")
+
+        # check string is the same as the file
+        with open('test.csv') as f:
+            # need to strip \r chars for windows
+            self.assertEqual(
+                csv_str.replace('\r', ''), f.read()
+            )
+
         # read csv
         df = pd.read_csv("test.csv")
         np.testing.assert_array_almost_equal(df["c"], solution.data["c"])
         np.testing.assert_array_almost_equal(df["2c"], solution.data["2c"])
 
+        # to json
+        solution.save_data("test.json", to_format="json")
+        json_str = solution.save_data(to_format="json")
+
+        # check string is the same as the file
+        with open('test.json') as f:
+            # need to strip \r chars for windows
+            self.assertEqual(
+                json_str.replace('\r', ''), f.read()
+            )
+
+        # check if string has the right values
+        json_data = json.loads(json_str)
+        np.testing.assert_array_almost_equal(json_data["c"], solution.data["c"])
```

Comment:
why is it only almost equal?

Reply:
I was only copying the other lines of this test. But it would need to be almost equal, as the numbers (e.g. 1/3) will be rounded off when they are written to json